features/shop/checkout/paying_for_order/preventing_payment_step_completion_without_a_selected_method.feature

```diff
@@ -10,16 +10,34 @@ Feature: Preventing payment step completion without a selected method
         And the store has a product "PHP T-Shirt" priced at "$19.99"
         And I am a logged in customer
 
-    @todo-api @ui @javascript
+    @api @no-ui
+    Scenario: Preventing payment step completion if there are no available methods
+        Given I have product "PHP T-Shirt" in the cart
+        And I have addressed the cart to "United States"
+        And I proceed with selecting "Free" shipping method
+        When I check the details of my cart
+        Then I should see that no payment method is assigned
+        And there should not be any payment methods available for selection
+
+    @no-api @ui @javascript
     Scenario: Preventing payment step completion if there are no available methods
         Given I have product "PHP T-Shirt" in the cart
         When I specified the billing address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
         And I proceed with selecting "Free" shipping method
-        And I do not select any payment method
         Then I should not be able to complete the payment step
         And there should be information about no payment methods available for my order
 
-    @todo-api @ui @javascript
+    @api @no-ui
+    Scenario: Preventing payment step completion if there are no available methods for a channel
```

Comment:
If I understand correctly we are separating the scenarios, becouse of addressing step, arent we able to use the AddressFactory to unify the steps on api side?